### DISCUSSION TITLE
Multitrack audio

### DISCUSSION THREAD

**Discussion by danivanderbeek** [katspaugh/wavesurfer.js#2739] [Ideas]
Hello,

I'm looking for a way to make Wavesurfer.js multitrack. As of right now, it is not, there is wavesufer playlist, but it is outdated and does not support plugins like this awesome project does.

Are there any plans to make it support multi track? As in, multiple audio files?

**Comment by katspaugh** [comment]
I’d say this is something that should be coded on the application level rather than in the library. Multitrack audio is a very broad requirement.

**Reply by danivanderbeek** [reply]
Yes, this is broad, but the idea is to accept multiple audio tracks, rendered under each other and interacting. So let's say you have track 1 and track 2 overlaying from let's say 200 seconds into the first track. See image as reference.

![MixEditor564](https://user-images.githubusercontent.com/49678036/227022580-1f9e3901-e1cf-4471-b6d7-33e890c1642a.jpeg)

If someone is able to implement this, I'd be willing to pay for the implementation, or help getting it done with the current possibilites of the project. The thing is, I'd love to build it myself, but my js knowledge is not sufficient as I'm more of a PHP developer with very basic JS skills.

Edit: don't look at all the controls, image is only to provide an idea of what I mean by multitrack. It also does not have to be "unlimited tracks".

**Reply by katspaugh** [reply]
I see what you mean.
This would require aligning the tracks between each other, so that they have the same time scale and an offset.
It can be done by adjusting their audio buffers in Web Audio. Basically pad with zeroes from the left for an offset, and from the right to make them the same length/duration.
Feel free to write me at katspaugh@gmail.com, we can figure something out.

**Reply by danivanderbeek** [reply]
Thank you, I'll shoot you a message today
